azurelinuxagent/common/cgroup.py

```diff
@@ -27,7 +27,8 @@
 
 AGENT_NAME_TELEMETRY = "walinuxagent.service"  # Name used for telemetry; it needs to be consistent even if the name of the service changes
 
-MetricValue = namedtuple('Metric', ['category', 'counter', 'instance', 'value'])
+MetricValue = namedtuple('Metric', ['category', 'counter', 'instance', 'value', 'report_period'])
+MetricValue.__new__.__defaults__ = (None,)  # namedtuple() assigns the values in the defaults iterable to the rightmost fields
```

Comment:
what is the motivation for defaulting to None? explicit initialization may be a better option

Reply:
This default only for report_period field. A very fee candidates (Metrics) have frequency set and rest don't have it. I thought explicit initialization with default on every metric is overhead when compared to this. This is like a class with default values for instance variables.

Reply:
defaulting to None makes it easier to introduces bugs. let's do explicit initialization

Reply:
Quite not sure If I followed, are you suggesting explicit initialization means other than None?